src/Symfony/Component/Form/FormFactory.php

```diff
@@ -126,7 +126,7 @@ public function createBuilderForProperty($class, $property, $data = null, array
 
         // user options may override guessed options
         if ($typeGuess) {
-            $options = array_merge($typeGuess->getOptions(), $options);
+            $options = array_replace_recursive($typeGuess->getOptions(), $options);
```

Comment:
I am not sure if it is a good idea to apply this "globally". For the `attr` option, doing a recursive merge does work. But this may not be the case for all other options where the usage of some keys may be mutually exclusive.